### ISSUE TITLE
Token expiration handler not working

### ISSUE THREAD

**Issue by gentlemanjohn** [auth0/socketio-jwt#53]
On my server I have:
```
var server = http.createServer(app);
var io = require('socket.io')(server);
var socketio_jwt = require('socketio-jwt');

io.sockets

  .on('connection', socketio_jwt.authorize({
    secret: app.get('superSecret')
  }))

  .on('authenticated', function(socket) {
    //this socket is authenticated, we are good to handle more events from it.
    console.log('+++++++++++++++++++++++++++++++++++++++++++++');
    console.log(socket.decoded_token.username, 'authenticated!');
    console.log('+++++++++++++++++++++++++++++++++++++++++++++');
  });
```

On my client (Ionic/Angular) I have:
```
    Socket.on('connect', function() {
      console.log('connected!');
    });

    Socket.on('authenticated', function () {
      console.log('authenticated!');
    });

    Socket.emit('authenticate', {
      token: $localStorage.auth_token
    });
            
    Socket.on('error', function(error) {
      console.log(error);
      if (error.type == "UnauthorizedError" || error.code == "invalid_token") {
        console.log("User's token has expired");
      }
```

Connection and authentication work as expected (client console shows 'connected!', 'authenticated!' on log in), however the error never fires on the client when the token expires (set to 1 minute).  What am I missing?

**Comment by gentlemanjohn** [comment]
Peeked into the souce and found that `Socket.on('error', ...` should be `Socket.on('unauthorized',...` on my client. The example should be updated.

**Comment by DiknT** [comment]
also from the example it's not 'error.type' OR 'error.code' but 'error.data.type' and 'error.data.code'. Full example should be updated to:
```javascript
socket.on('unauthorized', function (error) {
  if (error.data.type == "UnauthorizedError" || error.data.code == "invalid_token") {
    // redirect user to login page perhaps?
    console.log("User's token has expired");
  }
});
```

**Comment by mypark** [comment]
Confusingly, the original syntax of on "error" works if using the single roundtrip example   

```
io.use(socketioJwt.authorize({
    secret: 'test',
    handshake: true
  }));
```

Why is it handled differently between the 2 scenarios?

**Comment by gjsmith66** [comment]
That is because using the single round trip it does not allow Socket.IO to complete the connection, so Socket.IO emits the error. On the second round-trip example, Socket.IO does connect, however since unauthorized, socketio.jwt is emits the response, in this case unauthorized.

**Comment by jfromaniello** [comment]
I just merged #66, thanks and sorry for the confusion
